src/EditorFeatures/CSharpTest/CodeActions/SyncNamespace/SyncNamespaceTests_ChangeNamespace.cs

```diff
@@ -71,28 +168,147 @@ public async Task ChangeNamespace_SingleDocumentNoReference()
             var declaredNamespace = "Foo.Bar";
 
             var (folder, filePath) = CreateDocumentFilePath(new[] { "B", "C" }, "File1.cs");
-            var code =
-$@"
-<Workspace>
-    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" RootNamespace=""{defaultNamespace}"" CommonReferences=""true"">
-        <Document Folders=""{folder}"" FilePath=""{filePath}""> 
-namespace [||]{declaredNamespace}
-{{
+
+            var code = $@"
+namespace $${declaredNamespace}
+{{    
     class Class1
     {{
     }}
-}}</Document>
-    </Project>
-</Workspace>";
-
-            var expectedSourceOriginal =
-@"namespace A.B.C
+}}  
+";
+            var expected = @"namespace A.B.C
 {
     class Class1
     {
     }
-}";
-            await TestChangeNamespaceAsync(code, expectedSourceOriginal);
+}
+";
+            await new VerifyCS.Test
+            {
+                ProjectFilePath = ProjectFilePath,
+                RootNamespace = defaultNamespace,
+                TestState =
+                {
+                    Sources =
+                    {
+                        (PathUtilities.CombinePaths(PathUtilities.CombinePaths(ProjectRootPath, folder), filePath), code),
+                    },
+                },
+                FixedState =
+                {
+                    Sources =
+                    {
+                        (PathUtilities.CombinePaths(PathUtilities.CombinePaths(ProjectRootPath, folder), filePath), expected),
+                    },
+                },
+            }.RunAsync();
+        }
+
+        [WpfTheory, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        [CombinatorialData]
+        public async Task ChangeNamespace_AddUsingsPlacement(
+            [CombinatorialValues(AddImportPlacement.OutsideNamespace, AddImportPlacement.InsideNamespace)] Enum addImportPlacement)
+        {
+            var defaultNamespace = "A";
+
+            var (folder1, filePath1) = CreateDocumentFilePath(new[] { "B", "C" }, "File1.cs");
+            var (folder2, filePath2) = CreateDocumentFilePath(new[] { "B", "C" }, "File2.cs");
+
+            var code1 = @"
+namespace $$Foo.Bar
+{
+    class Class1 : Class2
+    {
+    }
+}
+";
+            var code2 = @"
+namespace Foo.Bar
+{
+    class Class2
+    {
+        Class1 Derived;
+    }
+}
+";
+
+            var expectedOutside1 = @"
+using Foo.Bar;
+
+namespace A.B.C
+{
+    class Class1 : Class2
+    {
+    }
+}
+";
+            var expectedOutside2 = @"
+using A.B.C;
+
+namespace Foo.Bar
+{
+    class Class2
+    {
+        Class1 Derived;
+    }
+}
+";
+
+            var expectedInside1 = @"
+namespace A.B.C
+{
+    using Foo.Bar;
```

Comment:
📝 This portion is failing because the refactoring always adds the using directive to the outer container, even when the user indicates the namespaces belong inside.